test/test.js

```diff
@@ -472,6 +472,25 @@ describe('options', function() {
         }, 100);
       });
     });
+
+    it('should be able to skip subdirectories', function(done) {
+      var options = {
+        delay: 0,
+        recursive: true,
+        filter: function(name, skip) {
+          if (/deep_node_modules/.test(name)) return skip;
+        }
+      };
+
+      watcher = watch(tree.getPath('home'), options);
+
+      watcher.getWatchedPaths(function(paths) {
+        var found = paths.find(n => /deep_node_modules/.test(n));
```

Comment:
Is the behaviour consistent and predictable on all systems? If so, then it might be clearer to perform a deep equal on `paths` on what is being watched, instead of only on what is not being watched, which feels like it could hide issues.

Reply:
I think it's not consistent with `recursive` option.  On linux it will return the root and all its subdirectories, while on MacOS it will only include the root directory since the `recursive` is supported natively.

I can't find a better name for `getWatchedPaths', maybe `getScannedPaths` might be more accurate?

Reply:
Hmmm `getWatchedPaths` should be tested separately, and I'll export `hasNativeRecursive` to use it in test for different situations. I'll do that later today.